### ISSUE TITLE
`versionReconciliation` ignores `dependencyOverrides`

### ISSUE THREAD

**Issue by jto** [coursier/coursier#1523]
## Expected Behavior

`versionReconciliation` should not issue an error for artifact versions forced by `dependencyOverrides`

## Current Behavior

When using `versionReconciliation` in a sbt build (either with `sbt-coursier` or `sbt-lm-coursier`), coursier will issue a conflict error even though the conflicting dependency version is forced by `dependencyOverrides`.

See the following sbt session (tested in one of [Scio](https://github.com/spotify/scio)'s submodules:

```sbt
sbt:scio-bigquery> show versionReconciliation
[info] * io.grpc:grpc-stub:strict

sbt:scio-bigquery> show dependencyOverrides
[info] * io.grpc:grpc-core:1.17.1
[info] * io.grpc:grpc-context:1.17.1
[info] * io.grpc:grpc-auth:1.17.1
[info] * io.grpc:grpc-netty:1.17.1
[info] * io.grpc:grpc-stub:1.17.1
[info] * com.google.guava:guava:25.1-jre
[info] * com.google.api:gax:1.38.0
[info] * com.google.api:gax-grpc:1.38.0

sbt:scio-bigquery> update
[error] stack trace is suppressed; run last scio-core / coursierResolutions for the full output
[error] (scio-core / coursierResolutions) coursier.error.conflict.StrictRule: Rule Strict(Set(ModuleMatcher(io.grpc:grpc-stub)),Set(),false,true,false) not satisfied: coursier.params.rule.Strict$EvictedDependencies: Unsatisfied rule Strict(io.grpc:grpc-stub): Found evicted dependencies:
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1beta1:0.44.0
[error]    └─ com.google.cloud:google-cloud-bigquerystorage:0.79.0-alpha
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigtable-admin-v2:0.38.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigtable-v2:0.38.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-pubsub-v1:1.43.0
[error]    └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]       └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-admin-database-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-admin-instance-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.10.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-common-protos:1.12.0
[error]    ├─ com.google.cloud:google-cloud-spanner:1.6.0
[error]    │  └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]    │     └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud:google-cloud-bigtable:0.73.0-alpha
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud:google-cloud-bigtable-admin:0.73.0-alpha
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]    └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]       └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigquerystorage-v1beta1:0.44.0
[error]    └─ com.google.cloud:google-cloud-bigquerystorage:0.79.0-alpha
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigtable-admin-v2:0.38.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-bigtable-v2:0.38.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-pubsub-v1:1.43.0
[error]    └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]       └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-admin-database-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-admin-instance-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-cloud-spanner-v1:1.6.0
[error]    └─ com.google.cloud:google-cloud-spanner:1.6.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.10.1 wanted)
[error] └─ com.google.api.grpc:grpc-google-common-protos:1.12.0
[error]    ├─ com.google.cloud:google-cloud-spanner:1.6.0
[error]    │  └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]    │     └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud:google-cloud-bigtable:0.73.0-alpha
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud:google-cloud-bigtable-admin:0.73.0-alpha
[error]    └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]       └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]          └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
[error]
[error] io.grpc:grpc-stub:1.17.1 (1.16.1 wanted)
[error] └─ com.google.cloud.bigtable:bigtable-client-core:1.8.0
[error]    └─ org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.18.0
[error]       └─ org.apache.beam:beam-runners-google-cloud-dataflow-java:2.18.0
```

## Context
- sbt `1.3.7` 
- `sbt-lm-coursier` or `sbt-coursier` version `2.0.0-RC5-3`

**Comment by alexarchambault** [comment]
There are a number of known issues around `dependencyOverrides` in sbt-coursier, it's possible that it's simply ignored currently (which has to be fixed too).

**Comment by alexarchambault** [comment]
@jto I couldn't [reproduce that](https://github.com/alexarchambault/coursier-issue-1523), neither in sbt or via the coursier CLI.

I did spot an issue with `versionReconciliation`, that was enabling strict reconciliation for all dependencies. It was making resolution fail for `scio-bigquery`, at commit `ffabbc3fcf7` in scio, when adding
```scala
versionReconciliation += "io.grpc" % "grpc-stub" % "strict"
```
to `scio-bigquery`. https://github.com/coursier/sbt-coursier/pull/244 should fix that.

**Comment by jto** [comment]
I can reproduce it with the following example:

```scala
scalaVersion := "2.12.11"
libraryDependencies ++= Seq(
  "com.spotify" %% "scio-core" % "0.8.4"
)
versionReconciliation += "io.grpc" % "grpc-stub" % "strict" //OK if commented, fails otherwise.

dependencyOverrides += "io.grpc" % "grpc-stub" % "1.17.1"
```

**Comment by alexarchambault** [comment]
Cool, I was able to reproduce that, and I confirm that https://github.com/coursier/sbt-coursier/pull/244 fixes it!

**Comment by alexarchambault** [comment]
Closing this, this should be fixed in the upcoming `2.0.0-RC6-6`, via https://github.com/coursier/sbt-coursier/pull/244.
